### PR TITLE
Alex Noto

### PR DESCRIPTION
**Pull request by noto-alex-j**

### CODE REVIEW

File: lib/arrays.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ There are multiple ways to add elements to an array. Elements of one array can b
 
 # Given the Array `["Laura", "Fiona", "Tori"]`, how would you replace `"Fiona"` with `"Florence"` so that you end up with `["Laura", "Florence", "Tori"]`?
 
-The simplest way I've found so far to replace an element in an array is to iterate through the array with the .each method, and call the .sub method on each element. For example, for the above array, I would write array.each do |x| x.sub("Fiona", "Florence") end. 
+A value in an array can be changed in a way that is similar to looking up the value using its index. In the above array, array[1] would return "Fiona", and array[1] = "Florence" would reassign the value of "Florence" to the index 1. I actually discovered this while playing with replacing elements in hashes. The way I had initially found to replace an element in an array was to iterate through the array with the .each method, and call the .sub method on each element. For example, for the above array, I would write array.each do |x| x.sub("Fiona", "Florence") end.
```
Comment (halfghaninne) [inline lib/arrays.md:32]: Does using `sub` permanently alter the original array (that is, if you were to add the line `puts array` at the end of your program, will you see `["Laura", "Florence", "Tori"]`)? If not, how might you slightly alter your `each` block to permanently change the original array?

File: lib/arrays.md [hunk 2]
```diff
@@ -5,44 +5,44 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
...
 
 # What is one way, using Ruby, to retrieve the 6th element in an Array? How about the 8th element? What happens if you try to retrieve the value of the _9999th_ element (or some element that doesn't exist in the array)?
 
-Replace me with your response...
+Each element in an array is assigned an index number, starting with 0, and an element can be retrieved by using that index. For example, if an_array = ["blue", "red", "yellow", "green", "purple", "orange", "black", "white"], then print an_array[5] will return "orange". The same would be true for the 8th element, print an_array[7] would return "white". Because there is no 50th element, calling print an_array[50] will return nil.
 
 # The previous question asks about finding, for example, the 6th element in an Array. Is it possible to find the **-6th** (Notice the negative symbol!) element in an Array? What does that even mean?
 
-Replace me with your response...
+It is possible to find a negative index in an array. If a negative index is accessed, the indexing order is reversed and the last value is assigned 0 rather than the first. So calling print an_array[-5] would return "yellow".
```
Comment (halfghaninne) [inline lib/arrays.md:20]: Close! When moving backward through an Array, indices start at `-1` (that way there can't be two elements at index `0`). Calling`an_array[-5]` would return `"green"` and `an_array[-6]` would return `"yellow"`.

File: lib/arrays.md (hunk 1 above)
Comment (noto-alex-j) [inline lib/arrays.md:32]: From what I understand, using .sub! will permanently alter the original array.

File: lib/arrays.md (hunk 2 above)
Comment (noto-alex-j) [inline lib/arrays.md:20]: Oh good to know, that makes total sense!

**Comment by noto-alex-j** [comment]
Answered question!

File: lib/arrays.md (hunk 1 above)
Comment (halfghaninne) [inline lib/arrays.md:32]: 👍

**Comment by halfghaninne** [comment]
Thanks! Nice work! 👍

**Pull request closed without merging**